### PR TITLE
[OPL-01] Favorite list and button feature added

### PR DESCRIPTION
**Pull request by opolancoh-gl**
Pending to integrate with Podcasts List

### CODE REVIEW

File: src/components/favorite-list/favorite-list.vue [hunk 1]
```diff
@@ -0,0 +1,53 @@
+<!--<template>
+  <List :title="'Favorite Podcasts'"  :list="podcasts" />
+</template>
+
+<script src="./script.js"></script>-->
+
+<template>
```
Comment (alejandrobenjumeaGL) [inline src/components/favorite-list/favorite-list.vue:1]: Is this a copy of **list-podcast** component? why not use that component?

**Review by alejandrobenjumeaGL — CHANGES_REQUESTED** [review]
I'm not seeing API integration with favorite feature

File: src/components/favorite-button/favorite-button.vue [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  <v-tooltip left>
+    <template v-slot:activator="{ on }">
+      <div @click="handleSetFavorite" v-on="on">
+        <v-icon v-if="favorite" color="yellow">mdi-star</v-icon>
+        <v-icon v-else color="grey lighten-1">mdi-star-outline</v-icon>
+      </div>
+    </template>
+    <span>{{tooltipText}}</span>
+  </v-tooltip>
+</template>
```
Comment (josepenagosGL) [inline src/components/favorite-button/favorite-button.vue]: Please recall we're not using single file components, must split the code into three different files

File: src/components/favorite-button/favorite-button.vue [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      required: true,
+    },
+  },
+
+  data: () => ({
+    favorite: null,
+  }),
+
+  created() {
+    this.favorite = this.isFavorite;
```
Comment (josepenagosGL) [inline src/components/favorite-button/favorite-button.vue]: Is this necessary? Or can we just use a prop and make it reactive?

**Review by josepenagosGL — CHANGES_REQUESTED** [review]

File: src/components/favorite-button/favorite-button.vue [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    <span>{{tooltipText}}</span>
+  </v-tooltip>
+</template>
+
+<script>
+export default {
+  name: 'FavoriteButton',
+
+  props: {
+    trackId: {
```
Comment (alejandrobenjumeaGL) [inline src/components/favorite-button/favorite-button.vue]: This component could work with other favorites scope, not only for tracks, so I suggest removing track prefix, just **id**

File: src/components/favorite-button/favorite-button.vue [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+  computed: {
+    tooltipText() {
+      return this.favorite ? 'Starred' : 'Not starred';
+    },
+  },
+
+  methods: {
+    handleSetFavorite() {
+      this.favorite = !this.favorite;
```
Comment (alejandrobenjumeaGL) [inline src/components/favorite-button/favorite-button.vue]: Is this managed state? this should reactive as @josepenagosGL said above.

**Review by alejandrobenjumeaGL — CHANGES_REQUESTED** [review]

File: src/components/favorite-list/favorite-list.vue (hunk 1 above)
Comment (opolancoh-gl) [reply src/components/favorite-list/favorite-list.vue:1]: My favorite-list component will call the podcast-list component, and that code is commented on the top. When @josepenagosGL integrates the favorite-button on his component, I'll uncomment that code and delete all the code remaining.

File: src/components/favorite-button/favorite-button.vue (hunk 2 above)
Comment (opolancoh-gl) [reply src/components/favorite-button/favorite-button.vue]: It's too short to split it into multiple files, isn't it? But I can do if necessary.

Comment (alejandrobenjumeaGL) [reply src/components/favorite-button/favorite-button.vue]: This is a defined standard, we agreed this directive  on the first day

**Pull request closed without merging**